src/Bicep.Core/Semantics/SymbolHelper.cs

```diff
@@ -63,6 +63,35 @@ objectType.AdditionalProperties is null ||
 
             switch (syntax)
             {
+                case StringSyntax stringSyntax:
```

Comment:
Would it be possible to update the [predicate function in BicepSymbolResolver](https://github.com/Azure/bicep/blob/d85be472671be2e98992acc2800173d9febbb6f0/src/Bicep.LangServer/Providers/BicepSymbolResolver.cs#L43) rather than adding special handling for string keys here? There are already blocks for handling both `ObjectPropertySyntax` and `ObjectPropertyTypeSyntax`, so having a dedicated block for `StringSyntax` leads to some duplication.